GAE/src/com/gallatinsystems/survey/dao/SurveyAssignmentDAO.java

```diff
@@ -16,13 +16,52 @@
 
 package com.gallatinsystems.survey.dao;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.waterforpeople.mapping.domain.SurveyAssignment;
 
+import com.gallatinsystems.common.Constants;
 import com.gallatinsystems.framework.dao.BaseDAO;
 
 public class SurveyAssignmentDAO extends BaseDAO<SurveyAssignment> {
 
     public SurveyAssignmentDAO() {
         super(SurveyAssignment.class);
     }
+    
+    /**
+     * Return a list of survey assignments containing a specified device
+     *
+     * @return list of assignments
+     */
+    public List<SurveyAssignment> listAllContainingDevice(Long devId) {
+    	List<SurveyAssignment> all = list(Constants.ALL_RESULTS);
```

Comment:
I think it's possible to get all assignments for a deviceId, including a screenshot from the console
![screenshot from 2019-01-24 18-10-22](https://user-images.githubusercontent.com/923280/51695492-774c7180-2003-11e9-84cf-def7a9e24050.png)

Reply:
Interesting, so the index generation looks inside lists. I never knew. Optimised that.